src/modules/RequestChain/native/URLPatternResourcesIndex.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Drash - A microframework for building JavaScript/TypeScript HTTP systems.
+ * Copyright (C) 2023  Drash authors. The Drash authors are listed in the
+ * AUTHORS file at <https://github.com/drashland/drash/AUTHORS>. This notice
+ * applies to Drash version 3.X.X and any later version.
+ *
+ * This file is part of Drash. See <https://github.com/drashland/drash>.
+ *
+ * Drash is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Drash is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * Drash. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// Imports > Modules
+import { ResourceClassesArray } from "../../base/types/ResourceClassesArray.ts";
+import { ResourcesIndex as BaseResourcesIndex } from "../../base/ResourcesIndex.ts";
+
+export class URLPatternResourcesIndex extends BaseResourcesIndex {
+  constructor(...resources: ResourceClassesArray) {
+    // @ts-ignore (crookse) Making the compiler happy
+    super(URLPattern, resources);
```

Comment:
nah, URLPattern follows the interface (the interface is based on URLPattern), but i ignored it because i wasnt using the deno extension lmao. now that you bring it up though, maybe we use a triple slash directive in files that use URLPattern?

Reply:
Oh, wait. This is because of the interoperability on your editor when using Deno vs Node? If so, can't we just not include this file in Node's tsconfig (include/exclude fields). That way when building the Node code, TypeScript will not account this file in the bundle, therefore it shouldn't error.

Reply:
kind of

> This is because of the interoperability on your editor when using Deno vs Node?

yes, but this is because i don't have Deno's extension enabled. my reasoning for this is i don't think we should care about what runtime we're building for. IMO, i think we should only care about what JS APIs we're building with and leave it to the user and runtime to determine if they can/cannot use it. if they can't use what we build and we're aware of it, we can publish a polyfill version (more on this below) or provide example code showing how someone would make a polyfill

> That way when building the Node code, TypeScript will not account this file in the bundle, therefore it shouldn't error.

i don't think i understand this part. the build process doesn't build for a specific runtime. the build process builds everything that we wrote in CJS and ESM formats.

correct me if i'm wrong, but we NEED this file with `URLPattern` so that it can be used by users in runtimes that do support `URLPattern`
- if `URLPattern` is not supported and the user wants the same functionality from this file, they can use the polyfill version of it: [src/modules/RequestChain/polyfill/URLPatternPolyfillResourcesIndex.ts](https://github.com/crookse/fork-drash/blob/breaking/v3-08-2023/src/modules/RequestChain/polyfill/URLPatternPolyfillResourcesIndex.ts)
  - this polyfill version is the nasty v1 implementation with regex paths and stuff
- if `URLPattern` is not supported and the user wants the same functionality from this file, but they DO NOT want to use the polyfill version that we provide, then they can:
  - use the base class ([](https://github.com/crookse/fork-drash/blob/breaking/v3-08-2023/src/standard/handlers/SearchIndex.ts)) and build out their own implementation (like using the path-to-regex library or urlpattern-polyfill library); or
  - extend the polyfill version we have and override some stuff